packages/react-components/src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import { ChevronDown } from '@livechat/design-system-icons';
+import cx from 'clsx';
+
+import { useAnimations } from '../../utils';
+import { Icon } from '../Icon';
+import { Text } from '../Typography';
+
+import { AccordionMultilineElement } from './components/AccordionMultilineElement';
+import { getLabel } from './helpers';
+import { IAccordionProps } from './types';
+
+import styles from './Accordion.module.scss';
+
+const baseClass = 'accordion';
+
+export const Accordion: React.FC<IAccordionProps> = ({
+  className,
+  children,
+  label,
+  multilineElement,
+  kind = 'default',
+  openOnInit = false,
+  isOpen,
+  onClose,
+  onOpen,
+  ...props
+}) => {
+  const isControlled = isOpen !== undefined;
+  const currentlyOpen = isControlled ? isOpen : openOnInit;
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const [size, setSize] = React.useState(0);
+  const previousSizeRef = React.useRef(size);
+  const {
+    isOpen: isExpanded,
+    isMounted,
+    setIsOpen,
+  } = useAnimations({
+    isVisible: currentlyOpen,
+    elementRef: contentRef,
+  });
+  const mergedClassName = cx(
+    styles[baseClass],
+    styles[`${baseClass}--${kind}`],
+    {
+      [styles[`${baseClass}--open`]]: isExpanded,
+    },
+    className
+  );
+
+  const handleStateChange = (state: boolean) => {
+    if (state) {
+      onClose?.();
+      !isControlled && setIsOpen(false);
+    } else {
+      onOpen?.();
+      !isControlled && setIsOpen(true);
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (!isExpanded && (event.key === 'Enter' || event.key === ' ')) {
+      handleStateChange(isExpanded);
+    }
+
+    if (isExpanded && event.key === 'Escape') {
+      handleStateChange(isExpanded);
+    }
+  };
+
+  React.useEffect(() => {
+    const hasIOSupport = !!window.ResizeObserver;
+
+    if (contentRef.current && hasIOSupport) {
+      const resizeObserver = new ResizeObserver((entries) => {
+        const entry = entries[0];
+        if (!entry) return;
+
+        const newSize = entry.contentRect.height;
+
+        if (previousSizeRef.current !== newSize) {
+          setSize(newSize);
+          previousSizeRef.current = newSize;
+        }
+      });
+
+      resizeObserver.observe(contentRef.current);
+
+      return () => resizeObserver.disconnect();
+    }
+  }, [contentRef]);
```

Comment:
Are you sure this effect works as intended? The contentRef is a reference so it's mutable and the useEffect between the renders will not detect its changes.